cmd/libsnap-confine-private/string-utils.c

```diff
@@ -69,3 +70,48 @@ int sc_must_snprintf(char *str, size_t size, const char *format, ...)
 
 	return n;
 }
+
+char *sc_must_stpcpy(char *buf, size_t buf_size, char *dest, const char *src)
+{
+	// Set errno in case we die.
+	errno = 0;
+	if (buf == NULL) {
+		die("cannot append string: buffer is NULL");
+	}
+	if (dest == NULL) {
+		die("cannot append string: destination is NULL");
+	}
+	if (src == NULL) {
+		die("cannot append string: source is NULL");
+	}
+	// Sanity check, the code doesn't need buffers larger than a few KBs so
+	// prevent corrupted or otherwise huge buffers from seeming "valid".
+	if (buf_size >= 0xFFFF) {
```

Comment:
What is this meant to protect against? The point of the function is to make sure that 'src + dst + \0' fits in buf_size. Why do we care how big buf_size is?